core/src/main/scala/com/raphtory/algorithms/temporal/Ancestors.scala

```diff
@@ -47,8 +47,18 @@ class Ancestors(seed: String, time: Long, delta: Long = Long.MaxValue, directed:
         if (vertex.name() == seed) {
           val edges = (if (directed) vertex.getInEdges() else vertex.getEdges())
             .filter(e => e.earliestActivity().time < time)
-            .filter(e => e.lastActivityBefore(time).time > time - delta)
-          edges.foreach(e => vertex.messageVertex(e.ID(), e.lastActivityBefore(time).time))
+            .filter(e =>
+              e.lastActivityBefore(time) match {
+                case Some(event) => event.time > time - delta
+                case None        => false
+              }
+            )
+          edges.foreach(e =>
+            e.lastActivityBefore(time) match {
+              case Some(event) => vertex.messageVertex(e.ID(), event.time)
+              case None        =>
+            }
+          )
           vertex.setState("ancestor", false)
```

Comment:
Actually, shouldn't this whole thing just be a foreach in the first place? It seem we are not using the `edge` value anywhere else?